boyar/main/main.go

```diff
@@ -104,27 +104,32 @@ func execute(flags *flags) {
 
 	if flags.daemonize {
 		<-supervized.GoForever(func() {
-			cfg, err := config.GetConfiguration(flags.configUrl, flags.ethereumEndpoint, flags.topologyContractAddress, flags.keyPairConfigPath)
-			if err != nil {
-				fmt.Println(time.Now(), "ERROR:", fmt.Errorf("could not generate configuration: %s", err))
-			} else {
-				reloadTimeDelay := cfg.ReloadTimeDelay(flags.maxReloadTimeDelay)
-				fmt.Println(fmt.Sprintf("INFO: waiting for %s to reload the configuration", reloadTimeDelay))
-				<-time.After(reloadTimeDelay)
-
-				ctx, cancel := context.WithTimeout(context.Background(), flags.timeout)
-				defer cancel()
-
-				if err := boyar.FullFlow(ctx, cfg, configCache); err != nil {
-					fmt.Println(time.Now(), "ERROR:", err)
+			for i := 0; true; i++ {
+				cfg, err := config.GetConfiguration(flags.configUrl, flags.ethereumEndpoint, flags.topologyContractAddress, flags.keyPairConfigPath)
+				if err != nil {
+					fmt.Println(time.Now(), "ERROR:", fmt.Errorf("could not generate configuration: %s", err))
+				} else {
+					// skip delay when provisioning for the first time when the node goes up
+					if i > 0 {
+						reloadTimeDelay := cfg.ReloadTimeDelay(flags.maxReloadTimeDelay)
+						fmt.Println(fmt.Sprintf("INFO: waiting for %s to apply new configuration", reloadTimeDelay))
+						<-time.After(reloadTimeDelay)
```

Comment:
also select on `ctx.Done()` so that sleeps of a few minutes won't delay termination

Reply:
This loop is not supposed to be terminated until the node is alive.